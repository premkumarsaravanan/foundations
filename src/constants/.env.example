MARKETPLACE_API_KEY_DEV=<<YOUR KEY>>
COGNITO_USERPOOL_ID_DEV=<<YOUR KEY>>
COGNITO_CLIENT_ID_DEV=<<YOUR KEY>>
