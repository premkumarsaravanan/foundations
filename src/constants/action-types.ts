/**
 * Please follow the <<STATE>>_<<ACTION_TYPE>> pattern and group actions by STATE
 */
const ActionTypes = {
  // Developer App Modal
  SET_DEVELOPER_APP_MODAL_STATE_EDIT_DETAIL: 'SET_DEVELOPER_APP_MODAL_STATE_EDIT_DETAIL',
  SET_DEVELOPER_APP_MODAL_STATE_VIEW_DETAIL: 'SET_DEVELOPER_APP_MODAL_STATE_VIEW_DETAIL',
  SET_DEVELOPER_APP_MODAL_STATE_DELETE: 'SET_DEVELOPER_APP_MODAL_STATE_DELETE',

  // Delete
  APP_DELETE_SET_INIT_FORM_STATE: 'APP_DELETE_SET_INIT_FORM_STATE',
  APP_DELETE_REQUEST: 'APP_DELETE_REQUEST',
  APP_DELETE_REQUEST_LOADING: 'APP_DELETE_REQUEST_LOADING',
  APP_DELETE_REQUEST_SUCCESS: 'APP_DELETE_REQUEST_SUCCESS',
  APP_DELETE_REQUEST_FAILURE: 'APP_DELETE_REQUEST_FAILURE',

  // Install
  APP_INSTALL_REQUEST_DATA: 'APP_INSTALL_REQUEST_DATA',
  APP_INSTALL_LOADING: 'APP_INSTALL_LOADING',
  APP_INSTALL_REQUEST_DATA_SUCCESS: 'APP_INSTALL_REQUEST_DATA_SUCCESS',
  APP_INSTALL_REQUEST_DATA_FAILURE: 'APP_INSTALL_REQUEST_DATA_FAILURE',
  APP_INSTALL_DONE: 'APP_INSTALL_DONE',

  // Uninstall
  APP_UNINSTALL_REQUEST_DATA: 'APP_UNINSTALL_REQUEST_DATA',
  APP_UNINSTALL_LOADING: 'APP_UNINSTALL_LOADING',
  APP_UNINSTALL_REQUEST_DATA_SUCCESS: 'APP_UNINSTALL_REQUEST_DATA_SUCCESS',
  APP_UNINSTALL_REQUEST_DATA_FAILURE: 'APP_UNINSTALL_REQUEST_DATA_FAILURE',
  APP_UNINSTALL_DONE: 'APP_UNINSTALL_DONE',

  // Auth actions
  AUTH_LOGIN: 'AUTH_LOGIN',
  AUTH_LOGIN_SUCCESS: 'AUTH_LOGIN_SUCCESS',
  AUTH_LOGIN_FAILURE: 'AUTH_LOGIN_FAILURE',
  AUTH_LOGOUT: 'AUTH_LOGOUT',
  AUTH_LOGOUT_SUCCESS: 'AUTH_LOGOUT_SUCCESS',
  AUTH_CHANGE_LOGIN_TYPE: 'AUTH_CHANGE_LOGIN_TYPE',
  AUTH_SET_REFRESH_SESSION: 'AUTH_SET_REFRESH_SESSION',
  AUTH_CLEAR: 'AUTH_CLEAR',

  // Error actions
  ERROR_THROWN_COMPONENT: 'ERROR_THROWN_COMPONENT',
  ERROR_THROWN_SERVER: 'ERROR_THROWN_SERVER',
  ERROR_CLEARED_COMPONENT: 'ERROR_CLEARED_COMPONENT',
  ERROR_CLEARED_SERVER: 'ERROR_CLEARED_SERVER',

  // Client actions
  CLIENT_REQUEST_DATA: 'CLIENT_REQUEST_DATA',
  CLIENT_REQUEST_FAILURE: 'CLIENT_REQUEST_FAILURE',
  CLIENT_LOADING: 'CLIENT_LOADING',
  CLIENT_RECEIVE_DATA: 'CLIENT_RECEIVE_DATA',
  CLIENT_CLEAR_DATA: 'CLIENT_CLEAR_DATA',

  // My apps actions
  MY_APPS_REQUEST_DATA: 'MY_APPS_REQUEST_DATA',
  MY_APPS_LOADING: 'MY_APPS_LOADING',
  MY_APPS_REQUEST_DATA_FAILURE: 'MY_APPS_REQUEST_DATA_FAILURE',
  MY_APPS_RECEIVE_DATA: 'MY_APPS_RECEIVE_DATA',
  MY_APPS_CLEAR_DATA: 'MY_APPS_CLEAR_DATA',

  // Developer actions
  DEVELOPER_REQUEST_DATA: 'DEVELOPER_REQUEST_DATA',
  DEVELOPER_REQUEST_DATA_FAILURE: 'DEVELOPER_REQUEST_DATA_FAILURE',
  DEVELOPER_LOADING: 'DEVELOPER_LOADING',
  DEVELOPER_RECEIVE_DATA: 'DEVELOPER_RECEIVE_DATA',
  DEVELOPER_CLEAR_DATA: 'DEVELOPER_CLEAR_DATA',
  DEVELOPER_CREATE: 'DEVELOPER_CREATE',
  DEVELOPER_SET_FORM_STATE: 'DEVELOPER_SET_FORM_STATE',
  DEVELOPER_SHOW_MODAL: 'DEVELOPER_SHOW_MODAL',

  // App Detail actions
  APP_DETAIL_REQUEST_DATA: 'APP_DETAIL_REQUEST_DATA',
  APP_DETAIL_LOADING: 'APP_DETAIL_LOADING',
  APP_DETAIL_REQUEST_DATA_FAILURE: 'APP_DETAIL_REQUEST_DATA_FAILURE',
  APP_DETAIL_RECEIVE_DATA: 'APP_DETAIL_RECEIVE_DATA',
  APP_DETAIL_CLEAR_DATA: 'APP_DETAIL_CLEAR_DATA',

  // Admin actions
  ADMIN_LOADING: 'ADMIN_LOADING',
  ADMIN_REQUEST_REVISIONS: 'ADMIN_REQUEST_REVISIONS',
  ADMIN_RECEIVE_REVISIONS: 'ADMIN_RECEIVE_REVISIONS',

  // submit app actions
  DEVELOPER_SUBMIT_APP: 'DEVELOPER_SUBMIT_APP',
  DEVELOPER_SUBMIT_APP_SET_FORM_STATE: 'DEVELOPER_SUBMIT_APP_SET_FORM_STATE',
  DEVELOPER_SUBMIT_APP_LOADING: 'DEVELOPER_SUBMIT_APP_LOADING',
  DEVELOPER_SUBMIT_APP_REQUEST_DATA: 'DEVELOPER_SUBMIT_APP_REQUEST_DATA',
  DEVELOPER_SUBMIT_APP_RECEIVE_DATA: 'DEVELOPER_SUBMIT_APP_RECEIVE_DATA',

  // submit revision acions
  DEVELOPER_SUBMIT_REVISION: 'DEVELOPER_SUBMIT_REVISION',
  DEVELOPER_SUBMIT_REVISION_SET_FORM_STATE: 'DEVELOPER_SUBMIT_REVISION_SET_FORM_STATE',

  // Client actions
  ADMIN_APPROVALS_REQUEST_DATA: 'ADMIN_APPROVALS_REQUEST_DATA',
  ADMIN_APPROVALS_REQUEST_FAILURE: 'ADMIN_APPROVALS_REQUEST_FAILURE',
  ADMIN_APPROVALS_LOADING: 'ADMIN_APPROVALS_LOADING',
  ADMIN_APPROVALS_RECEIVE_DATA: 'ADMIN_APPROVALS_RECEIVE_DATA',
  ADMIN_APPROVALS_CLEAR_DATA: 'ADMIN_APPROVALS_CLEAR_DATA',

  // Revision Detail actions
  REVISION_DETAIL_REQUEST_DATA: 'REVISION_DETAIL_REQUEST_DATA',
  REVISION_DETAIL_LOADING: 'REVISION_DETAIL_LOADING',
  REVISION_DETAIL_REQUEST_DATA__FAILURE: 'REVISION_DETAIL_REQUEST_DATA__FAILURE',
  REVISION_DETAIL_RECEIVE_DATA: 'REVISION_DETAIL_RECEIVE_DATA',
  REVISION_DETAIL_CLEAR_DATA: 'REVISION_DETAIL_CLEAR_DATA',

  REVISION_SUBMIT_APPROVE: 'REVISION_SUBMIT_APPROVE',
  REVISION_APPROVE_SET_FORM_STATE: 'REVISION_APPROVE_SET_FORM_STATE',
  REVISION_SUBMIT_DECLINE: 'REVISION_SUBMIT_DECLINE',
  REVISION_DECLINE_SET_FORM_STATE: 'REVISION_DECLINE_SET_FORM_STATE',

  // App detail modal
  SET_APP_DETAIL_MODAL_STATE_VIEW: 'SET_APP_DETAIL_MODAL_STATE_VIEW',
  SET_APP_DETAIL_MODAL_STATE_CONFIRM: 'SET_APP_DETAIL_MODAL_STATE_CONFIRM',
  SET_APP_DETAIL_MODAL_STATE_UNINSTALL: 'SET_APP_DETAIL_MODAL_STATE_UNINSTALL',
  SET_APP_DETAIL_MODAL_STATE_SUCCESS: 'SET_APP_DETAIL_MODAL_STATE_SUCCESS',

  // categories
  CATEGORIES_RECEIVE_DATA: 'CATEGORIES_RECEIVE_DATA',

  FORGOT_PASSWORD_SUBMIT_EMAIL: 'FORGOT_PASSWORD_SUBMIT_EMAIL',
  FORGOT_PASSWORD_LOADING: 'FORGOT_PASSWORD_LOADING',

  SETTING_FETCH_DEVELOPER_INFO: 'SETTING_FETCH_DEVELOPER_INFO',
  SETTING_FETCH_DEVELOPER_INFO_SUCCESS: 'SETTING_FETCH_DEVELOPER_INFO_SUCCESS',
  SETTING_SHOW_HIDE_LOADING: 'SETTING_SHOW_HIDE_LOADING',
  SETTING_UPDATE_DEVELOPER: 'SETTING_UPDATE_DEVELOPER',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD'
}

export default ActionTypes
