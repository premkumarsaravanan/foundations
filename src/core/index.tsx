import '../styles/index.scss'
import * as React from 'react'
import { render } from 'react-dom'
import Router from './router'
import { Provider } from 'react-redux'
import store from './store'
import Toast from '../components/ui/toast'
import { PortalProvider } from '@reapit/elements'

const rootElement = document.querySelector('#root') as Element

const App = () => (
  <Provider store={store.reduxStore}>
    <PortalProvider>
      <Router />
    </PortalProvider>
    <Toast />
  </Provider>
)

if (rootElement) {
  render(<App />, rootElement)
}

export default App
