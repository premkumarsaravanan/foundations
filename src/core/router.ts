import { Router, Response, Request } from 'express'
import { loginUserApi } from '../api/session/login-user-session'
import { refreshUserSessionApi } from '../api/session/refresh-user-session'
import { changePasswordApi } from '../api/password/change-password'
import errorStrings from '../constants/error-strings'
import errorHandler from '../utils/error-handler'

const router = Router()

router.post('/login', (req, res) => loginUserApi(req, res))
router.post('/refresh', (req, res) => refreshUserSessionApi(req, res))
router.post('/password/change', (req, res) => changePasswordApi(req, res))
router.post('/password/reset', (req, res) => changePasswordApi(req, res))
router.post('/password/confirm', (req, res) => changePasswordApi(req, res))
router.use((err: Error, _req: Request, res: Response) => {
  console.error(errorStrings.NOT_FOUND, JSON.stringify(err))
  errorHandler(res, 404, errorStrings.NOT_FOUND, err)
})

export default router
