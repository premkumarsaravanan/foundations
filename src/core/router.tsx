import * as React from 'react'
import { Route, Router as BrowserRouter, Switch, Redirect } from 'react-router-dom'
import Routes from '../constants/routes'
import PrivateRoute from './private-route'
import PrivateRouteWrapper from './private-route-wrapper'
import { createBrowserHistory } from 'history'
import { Info } from '@reapit/elements'

export const history = createBrowserHistory()

const Login = React.lazy(() => import('../components/pages/login'))
const Client = React.lazy(() => import('../components/pages/client'))
const MyApps = React.lazy(() => import('../components/pages/my-apps'))
const Register = React.lazy(() => import('../components/pages/register'))
const DeveloperHome = React.lazy(() => import('../components/pages/developer-home'))
const DeveloperSubmitApp = React.lazy(() => import('../components/pages/developer-submit-app'))
const AdminApprovalsPage = React.lazy(() => import('../components/pages/admin-approvals'))
const AdminDevManagementPage = React.lazy(() => import('../components/pages/admin-dev-management'))
const ApiDocsPage = React.lazy(() => import('../components/pages/api-docs'))
const SwaggerPage = React.lazy(() => import('../components/pages/swagger'))
const ElementsPage = React.lazy(() => import('../components/pages/elements'))
const DesktopDocsPage = React.lazy(() => import('../components/pages/desktop-api-docs'))
const WebComponentsPage = React.lazy(() => import('../components/pages/web-components'))
const SettingsPage = React.lazy(() => import('../components/pages/settings'))
// const AnalyticsPage = React.lazy(() => import('../components/pages/analytics'))
const ResetPassword = React.lazy(() => import('../components/pages/reset-password'))
const ForgotPassword = React.lazy(() => import('../components/pages/forgot-password/forgot-password'))
const AdminAppsPage = React.lazy(() => import('../components/pages/admin-apps'))

const Router = () => (
  <BrowserRouter history={history}>
    <React.Suspense fallback={null}>
      <Switch>
        <Route
          path={[Routes.CLIENT_LOGIN, Routes.DEVELOPER_LOGIN, Routes.ADMIN_LOGIN]}
          exact
          render={() => <Login />}
        />
        <Route path={Routes.DEVELOPER_RESET_PASSWORD} component={ResetPassword} />
        <Route path={Routes.REGISTER} exact render={() => <Register />} />
        <Route path={Routes.FORGOT_PASSWORD} component={ForgotPassword} />
        <Route path={Routes.FOUR_O_FOUR} exact render={() => <Info infoType="404" />} />
        <PrivateRouteWrapper path="/">
          <Switch>
            <PrivateRoute allow="CLIENT" path={Routes.MY_APPS_PAGINATE} component={MyApps} fetcher />
            <PrivateRoute allow="CLIENT" path={Routes.MY_APPS} component={MyApps} fetcher exact />
            <PrivateRoute allow="CLIENT" path={Routes.CLIENT} component={Client} exact fetcher />

            <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_MY_APPS} component={DeveloperHome} exact fetcher />
            <PrivateRoute
              allow="DEVELOPER"
              path={Routes.DEVELOPER_MY_APPS_PAGINATE}
              component={DeveloperHome}
              exact
              fetcher
            />
            <PrivateRoute
              allow="DEVELOPER"
              path={Routes.DEVELOPER_MY_APPS_EDIT}
              component={DeveloperSubmitApp}
              exact
              fetcher
            />
            <PrivateRoute allow="DEVELOPER" path={Routes.SUBMIT_APP} fetcher component={DeveloperSubmitApp} />
            <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_API_DOCS} exact component={ApiDocsPage} />
            <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_SWAGGER} exact component={SwaggerPage} />
            <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_ELEMENTS} exact component={ElementsPage} />
            {/* <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_ANALYTICS} exact component={AnalyticsPage} /> */}
            <PrivateRoute allow="DEVELOPER" path={Routes.DEVELOPER_DESKTOP_DOCS} exact component={DesktopDocsPage} />
            <PrivateRoute
              allow="DEVELOPER"
              path={Routes.DEVELOPER_WEB_COMPONENTS}
              exact
              component={WebComponentsPage}
            />
            <PrivateRoute allow="DEVELOPER" path={Routes.SETTINGS} fetcher exact component={SettingsPage} />

            <PrivateRoute allow="ADMIN" path={Routes.ADMIN_APPROVALS} component={AdminApprovalsPage} exact fetcher />
            <PrivateRoute allow="ADMIN" path={Routes.ADMIN_APPROVALS_PAGINATE} component={AdminApprovalsPage} fetcher />
            <PrivateRoute allow="ADMIN" path={Routes.ADMIN_APPS} component={AdminAppsPage} exact />
            <PrivateRoute
              allow="ADMIN"
              path={Routes.ADMIN_DEV_MANAGEMENT}
              component={AdminDevManagementPage}
              exact
              fetcher
            />
            <PrivateRoute
              allow="ADMIN"
              path={Routes.ADMIN_DEV_MANAGEMENT_PAGINATE}
              component={AdminDevManagementPage}
              exact
              fetcher
            />
            <Route render={() => <Info infoType="404" />} />
          </Switch>
        </PrivateRouteWrapper>
        <Redirect to={Routes.CLIENT_LOGIN} />
      </Switch>
    </React.Suspense>
  </BrowserRouter>
)

export default Router
