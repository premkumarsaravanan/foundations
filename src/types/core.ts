import Routes from '../constants/routes'
import ActionTypes from '../constants/action-types'
import { ClientState } from '../reducers/client'
import { MyAppsState } from '../reducers/my-apps'
import { DeveloperState } from '../reducers/developer'
import { AuthState } from '../reducers/auth'
import { AppDetailState } from '../reducers/app-detail'
import { ErrorState } from '../reducers/error'
import { AdminState } from '../reducers/admin'
import { SubmitAppState } from '../reducers/submit-app'
import { SubmitRevisionState } from '@/reducers/submit-revision'
import { AdminApprovalsState } from '@/reducers/admin-approvals'
import { RevisionDetailState } from '@/reducers/revision-detail'
import { AppPermissionState } from '@/reducers/app-permission'
import { AppDetailModalState } from '@/reducers/app-detail-modal'
import { AppInstallState } from '@/reducers/app-install'
import { AppUninstallState } from '@/reducers/app-uninstall'

export interface Action<T> {
  readonly type: ActionType
  readonly data: T
}

export interface ActionCreator<T> {
  readonly type: string
  (data: T): Action<T>
}

export interface StringMap {
  [key: string]: string
}

export type PartialRecord<K extends keyof any, T> = { [P in K]?: T }

export type RouteValue = keyof typeof Routes

export type ActionType = keyof typeof ActionTypes

export type FormState = 'PENDING' | 'DONE' | 'SUBMITTING' | 'ERROR' | 'SUCCESS'

export interface FetcherParams<T> {
  method: 'GET' | 'POST' | 'PUT' | 'DELETE'
  api: string
  url: string
  headers: StringMap
  isPrivate?: boolean
  body?: T
}

export interface ReduxState {
  client: ClientState
  myApps: MyAppsState
  appDetail: AppDetailState
  developer: DeveloperState
  auth: AuthState
  error: ErrorState
  admin: AdminState
  submitApp: SubmitAppState
  submitRevision: SubmitRevisionState
  adminApprovals: AdminApprovalsState
  revisionDetail: RevisionDetailState
  appPermission: AppPermissionState
  appDetailModal: AppDetailModalState
  appInstall: AppInstallState
  appUninstall: AppUninstallState
}
