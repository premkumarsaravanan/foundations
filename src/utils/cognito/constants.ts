export const COGNITO_API_BASE_URL = 'https://rbsbshnxvb.execute-api.eu-west-2.amazonaws.com/dev/api'

export const COGNITO_HEADERS = {
  'Content-Type': 'application/json'
}
