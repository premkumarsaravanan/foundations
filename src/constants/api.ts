import { StringMap } from '@/types/core'

export const CONTACTS_HEADERS = {
  'Content-Type': 'application/json'
} as StringMap

export const REAPIT_API_BASE_URL = 'https://reapit.cloud.tyk.io'

export const UPLOAD_FILE_BASE_URL = 'https://1qdnvftct6.execute-api.eu-west-2.amazonaws.com/dev/fileupload'

export const URLS = {
  contacts: '/contacts',
  idChecks: '/identitychecks',
  configuration: '/configuration'
}
