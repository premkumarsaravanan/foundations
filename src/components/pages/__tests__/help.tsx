import * as React from 'react'
import { shallow } from 'enzyme'
import {
  HelpPage,
  handleGotoWelcomeGuide,
  handleReportBug,
  handleRequestEndpoint,
  handleFaq,
  helpItems,
  renderHelpItems
} from '../help'
import Routes from '@/constants/routes'
import { history } from '@/core/router'
import { HelpLinks } from '@/constants/help-links'

jest.mock('../../../core/router', () => ({
  history: {
    push: jest.fn()
  }
}))

afterEach(() => {
  jest.clearAllMocks()
})

describe('HelpPage', () => {
  it('should match a snapshot', () => {
    expect(shallow(<HelpPage />)).toMatchSnapshot()
  })
})

describe('handleGotoWelcomeGuide', () => {
  it('should called with correct props', () => {
    const spy = jest.spyOn(history, 'push')
    handleGotoWelcomeGuide()
    expect(spy).toHaveBeenCalledWith(Routes.DEVELOPER_WELCOME)
  })
})

describe('handleReportBug', () => {
  it('should called with correct props', () => {
    const spy = jest.spyOn(window, 'open')
    handleReportBug()
    expect(spy).toHaveBeenCalledWith(HelpLinks.BUG_REPORT, '_blank')
  })
})

describe('handleRequestEndpoint', () => {
  it('should called with correct props', () => {
    const spy = jest.spyOn(window, 'open')
    handleRequestEndpoint()
    expect(spy).toHaveBeenCalledWith(HelpLinks.API_REQUEST, '_blank')
  })
})

describe('handleFaq', () => {
  it('should called with correct props', () => {
    const spy = jest.spyOn(window, 'open')
    handleFaq()
    expect(spy).toHaveBeenCalledWith(HelpLinks.FAQ, '_blank')
  })
})

describe('renderHelpItems', () => {
  it('should match snapshot', () => {
    expect(renderHelpItems(helpItems)).toMatchSnapshot()
  })
})
