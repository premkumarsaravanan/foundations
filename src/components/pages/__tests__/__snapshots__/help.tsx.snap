// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage should match a snapshot 1`] = `
<div>
  <H3>
    Help
  </H3>
  <GridFiveCol>
    <GridFourColItem
      key="Welcome Guide"
    >
      <FlexContainerBasic
        centerContent={true}
        flexColumn={true}
        hasPadding={true}
      >
        <div>
          <img
            alt="Welcome Guide"
            src=""
          />
        </div>
        <H4
          isCentered={true}
        >
          Welcome Guide
        </H4>
        <p>
          Need a little help? Have a look through the Welcome Guide, which we’ve put together to help you navigate through your Developer portal.
        </p>
        <p>
          <Button
            disabled={false}
            fullWidth={false}
            loading={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            VIEW
          </Button>
        </p>
      </FlexContainerBasic>
    </GridFourColItem>
    <GridFourColItem
      key="Request a Feature"
    >
      <FlexContainerBasic
        centerContent={true}
        flexColumn={true}
        hasPadding={true}
      >
        <div>
          <img
            alt="Request a Feature"
            src=""
          />
        </div>
        <H4
          isCentered={true}
        >
          Request a Feature
        </H4>
        <p>
          Use this form to request a feature in either the Marketplace or Foundations API. Please note, we will look at all requests carefully however, we cannot guarantee all will be implemented.
        </p>
        <p>
          <Button
            disabled={false}
            fullWidth={false}
            loading={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            REQUEST
          </Button>
        </p>
      </FlexContainerBasic>
    </GridFourColItem>
    <GridFourColItem
      key="Report a Bug"
    >
      <FlexContainerBasic
        centerContent={true}
        flexColumn={true}
        hasPadding={true}
      >
        <div>
          <img
            alt="Report a Bug"
            src=""
          />
        </div>
        <H4
          isCentered={true}
        >
          Report a Bug
        </H4>
        <p>
          Please report details of any bugs in relation to the Reapit Developer portal or Reapit Foundations API here. 
        </p>
        <p>
          <Button
            disabled={false}
            fullWidth={false}
            loading={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            REPORT
          </Button>
        </p>
      </FlexContainerBasic>
    </GridFourColItem>
    <GridFourColItem
      key="Need Help?"
    >
      <FlexContainerBasic
        centerContent={true}
        flexColumn={true}
        hasPadding={true}
      >
        <div>
          <img
            alt="Need Help?"
            src=""
          />
        </div>
        <H4
          isCentered={true}
        >
          Need Help?
        </H4>
        <p>
          If you need any support, we are here to help. Why not talk to one of our Developers or Product Owners directly.
        </p>
        <p>
          <Button
            disabled={false}
            fullWidth={false}
            loading={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            START CHAT
          </Button>
        </p>
      </FlexContainerBasic>
    </GridFourColItem>
  </GridFiveCol>
</div>
`;

exports[`renderHelpItems should match snapshot 1`] = `
<GridFiveCol>
  <GridFourColItem>
    <FlexContainerBasic
      centerContent={true}
      flexColumn={true}
      hasPadding={true}
    >
      <div>
        <img
          alt="Welcome Guide"
          src=""
        />
      </div>
      <H4
        isCentered={true}
      >
        Welcome Guide
      </H4>
      <p>
        Need a little help? Have a look through the Welcome Guide, which we’ve put together to help you navigate through your Developer portal.
      </p>
      <p>
        <Button
          disabled={false}
          fullWidth={false}
          loading={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          VIEW
        </Button>
      </p>
    </FlexContainerBasic>
  </GridFourColItem>
  <GridFourColItem>
    <FlexContainerBasic
      centerContent={true}
      flexColumn={true}
      hasPadding={true}
    >
      <div>
        <img
          alt="Request a Feature"
          src=""
        />
      </div>
      <H4
        isCentered={true}
      >
        Request a Feature
      </H4>
      <p>
        Use this form to request a feature in either the Marketplace or Foundations API. Please note, we will look at all requests carefully however, we cannot guarantee all will be implemented.
      </p>
      <p>
        <Button
          disabled={false}
          fullWidth={false}
          loading={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          REQUEST
        </Button>
      </p>
    </FlexContainerBasic>
  </GridFourColItem>
  <GridFourColItem>
    <FlexContainerBasic
      centerContent={true}
      flexColumn={true}
      hasPadding={true}
    >
      <div>
        <img
          alt="Report a Bug"
          src=""
        />
      </div>
      <H4
        isCentered={true}
      >
        Report a Bug
      </H4>
      <p>
        Please report details of any bugs in relation to the Reapit Developer portal or Reapit Foundations API here. 
      </p>
      <p>
        <Button
          disabled={false}
          fullWidth={false}
          loading={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          REPORT
        </Button>
      </p>
    </FlexContainerBasic>
  </GridFourColItem>
  <GridFourColItem>
    <FlexContainerBasic
      centerContent={true}
      flexColumn={true}
      hasPadding={true}
    >
      <div>
        <img
          alt="Need Help?"
          src=""
        />
      </div>
      <H4
        isCentered={true}
      >
        Need Help?
      </H4>
      <p>
        If you need any support, we are here to help. Why not talk to one of our Developers or Product Owners directly.
      </p>
      <p>
        <Button
          disabled={false}
          fullWidth={false}
          loading={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          START CHAT
        </Button>
      </p>
    </FlexContainerBasic>
  </GridFourColItem>
</GridFiveCol>
`;
